packages/material-ui/src/SwipeableDrawer/SwipeableDrawer.test.js

```diff
@@ -540,47 +508,40 @@ describe('<SwipeableDrawer />', () => {
             onClose={() => {}}
             open={false}
             PaperProps={{ component: FakePaper }}
+            SwipeAreaProps={{ 'data-testid': 'swipearea' }}
           >
             <div>Drawer2</div>
           </SwipeableDrawer>
         </div>,
       );
 
-      // use the same event object for both touch start events, one would propagate to the other swipe area in the browser
```

Comment:
Need to do some manual testing to understand what was replayed here.

1. events are never re-used
2. if this is about propagating then the DOM takes care of it (not dispatching multiple times). The original idea was probably replaying about hit detection

Reply:
Verified in https://codesandbox.io/s/single-swipearea-lock-ksyss (`monitorEvents(document.body, ['touchstart', 'touchmove', 'touchend'])`) that the way the test is setup a `touchstart` would only be fired at the last rendered `SwipeableDrawer. The `SwipeArea`s are stacked on top of each other and hit detection never "pierces" elements i.e. if a touch hits an element then the event is targetted at that element not at all the ones the touch pierces. Then the event bubbles up but since the `SwipeArea`s are siblings only one registers the `touchstart`.

I also remembered that subsequent `touchmove` and `touchend` target the same element as the `touchstart`.

But I don't know how I could get this test to fail. I thought https://github.com/mui-org/material-ui/blob/5f882fefadbdd72e950a4940a550c1d3287dd150/packages/material-ui/src/SwipeableDrawer/SwipeableDrawer.js#L300-L303 was responsible for this behavior but removing the early return does not fail any test.

@oliviertassinari Do you know what implementation this test was ensuring or how one would test https://github.com/mui-org/material-ui/blob/5f882fefadbdd72e950a4940a550c1d3287dd150/packages/material-ui/src/SwipeableDrawer/SwipeableDrawer.js#L300-L303

Also

Reply:
The new test case looks correct. Regarding getting it to fail. We added more defensive logic, later on, to avoid two swipeable drawers responding to the same event. It seems that this logic is winning over the previous solutions. For instance, `event.defaultMuiPrevented` now only seems to make sense if there is a custom component. `claimedSwipeInstance` now seems to be solved by testing the target of the event. We should be able to remove this logic.

Reply:
I'm somewhat leaning towards keeping the implementation to be safe. Touch related interactions are hard to test automatically so it may take a while before people realize we broken something.